fastapi/dependencies/_aliases.py

```diff
@@ -0,0 +1,42 @@
+from inspect import Parameter
+from typing import Any, Dict, Optional, Type, TypeVar, cast
+
+from fastapi.params import Deferred, Depends
+from fastapi.requests import Request
+from typing_extensions import Annotated, get_args, get_origin
+
+T = TypeVar("T")
+
+
+LifespanState = Dict[str, Any]
+
+
+def get_type(param: Parameter) -> Type[Any]:
+    annotation = param.annotation
+    while get_origin(annotation) is Annotated:
+        annotation = get_args(annotation)[0]
+    return annotation  # type: ignore[no-any-return]
+
+
+class _FromLifespan:
+    def __init__(self, param: Parameter) -> None:
+        self._type = get_type(param)
+        self._key: Optional[str] = None
```

Comment:
One interesting thing about this API is that there might be some type information available from the lifepsan itself. Should FastAPI introspect that? Should it force users to return something typed? Would that help prevent mistakes at startup, e.g. where the user requests a type that is not available or is maybe ambiguous?